### DISCUSSION TITLE
why is the `uri!` macro not parsing a route correctly?

### DISCUSSION THREAD

**Discussion by 2-3-5-41** [rwf2/Rocket#2453] [Questions] [answered]
I'm trying to use `uri!` with a redirect in my current learning project, and when I attempt to redirect to a route that I have in a module using the `uri!` macro causes rocket to return a 404 error, this is because the route should go to '/user/login' but the macro only resolves to '/login' and ignores the base path.

Here is the [repo](https://github.com/1SMOL/rocket-mongodb-user-auth), and the macro in question is being used in `src/public/user.rs'.

**Comment by wezm** [comment, marked as answer]
You're mounting your user routes with a prefix:

`.mount("/user", routes![`

You have to pass that same prefix to `uri!`

`uri!("/user", crate::public::user::login)`

There was some info about this decision in this comment:

https://github.com/SergioBenitez/Rocket/issues/1553#issuecomment-787523869

**Reply by 2-3-5-41** [reply]
I wish it wasn't easy for me to overlook things sometimes; I really need to slow down lol
